src/Chapter3.hs

```diff
@@ -973,6 +1051,28 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday = Sunday | Monday | Tuesday | Wednesday | Thursday | Friday | Saturday
+  deriving (Show, Read, Eq, Ord)
+
+isWeekend:: Weekday -> Bool 
+isWeekend day = day == Sunday || day == Saturday
+
+nextDay:: Weekday -> Weekday
+nextDay Sunday = Monday
+nextDay Monday = Tuesday
+nextDay Tuesday = Wednesday
+nextDay Wednesday = Thursday
+nextDay Thursday = Friday
+nextDay Friday = Saturday
+nextDay Saturday = Sunday
```

Comment:
That's a bit mouthful implementation 🙂 Though, it's correct 👍🏻 
But can you use functions of the `Enum` and `Bounded` typeclasses to implement this function succinctly?